### PR TITLE
Migrate configs for isort, mypy, and pytest into pyproject.toml

### PR DESCRIPTION
**Pull request by cclauss**
## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description of what the PR does.

## Description
Migrate the configs into `pyproject.toml` using [`ini2toml`](https://pypi.org/project/ini2toml) to do the file conversion and running [`validate-pyproject`](https://github.com/abravalheri/validate-pyproject) to validate the results.  Skip any configuration that is not yet PEP 621 compatible.
```
git checkout -b more-pyproject.toml
ini2toml setup.cfg
code pyproject.toml setup.cfg  # Cherry-pick those changes which are PEP 621 compatible.
validate-pyproject pyproject.toml
```
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Very nice, thank you !

**Comment by Pierre-Sassoulas** [comment]
I realized that ini2toml do not translate pylintrc to toml but it could :)

**Comment by cclauss** [comment]
`ini2toml pylintrc` generates output.

**Comment by Pierre-Sassoulas** [comment]
Would you mind also doing the pylintrc ?

**Comment by cclauss** [comment]
Can we please do that in a separate pull request?

**Review by Pierre-Sassoulas — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -54,3 +54,41 @@ include = ["astroid*"]
...
+
+[tool.mypy]
+enable_error_code = "ignore-without-code"
+no_implicit_optional = true
+scripts_are_modules = true
+show_error_codes = true
+warn_redundant_casts = true
+
+[[tool.mypy.overrides]]
+# Importlib typeshed stubs do not include the private functions we use
```
Comment (DanielNoord) [inline pyproject.toml:82]: I would have preferred to keep this comment on L86 so we know what it is referring to

**Pull request merged**